### PR TITLE
Remove deprecated features for release 0.13

### PR DESCRIPTION
**Pull request by bdice**
## Description
Removes deprecated code marked for removal in 0.13.0.

## Types of Changes

- [ ] Documentation update
- [ ] Bug fix
- [x] New feature
- [x] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:
- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-flow/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac-flow/blob/master/contributors.yaml).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md#code-style) of this project.
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [x] I have updated the API documentation as part of the package doc-strings.
- [ ] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [x] I have updated the [changelog](https://github.com/glotzerlab/signac-flow/blob/master/changelog.txt).

### CODE REVIEW

File: flow/project.py [hunk 1]
```diff
@@ -687,6 +610,9 @@ def __call__(self, *jobs, **kwargs):
             raise ValueError(f"Invalid keyword arguments: {', '.join(kwargs)}")
 
         if job is not None:
+            # TODO 0.13: How to remove this keyword argument? How should
+            # FlowCmdOperation act if multiple jobs are provided? Need to
+            # update both documentation and behavior.
```
Comment (bdice) [inline flow/project.py:615]: I'd like input on this, @glotzerlab/signac-committers. I don't know how to proceed.

Comment (bdice) [reply flow/project.py:615]: Maybe we change the end of the function to:
```python
cmd = self._cmd(*jobs) if callable(self._cmd) else self._cmd
format_arguments = {"jobs": jobs}
if len(jobs) == 1:
    format_arguments["job"] = jobs[0]
return cmd.format(**format_arguments)
```

Comment (kidrahahjo) [reply flow/project.py:615]: Can we advise users to not use the "job" variable in the command? Using only curly braces would make things simple, right?

Comment (bdice) [reply flow/project.py:615]: We should not break existing cmd operations. Typically a user provides a string like `my_executable --input {job.ws}/file.txt`. In that case we need to keep `job` as a format argument.

Comment (klywang) [reply flow/project.py:615]: What would be an example use for this?

> Maybe we change the end of the function to:
> 
> ```python
> cmd = self._cmd(*jobs) if callable(self._cmd) else self._cmd
> format_arguments = {"jobs": jobs}
> if len(jobs) == 1:
>     format_arguments["job"] = jobs[0]
> return cmd.format(**format_arguments)
> ```

Is this assuming that, if someone provides multiple jobs, they would expect it to execute as something like `my_executable --input job1 job2 job3 --another_flag`? 

And would the alternative expected behavior is to have `my_executable --input {} --another_flag` (or whatever the provided shell command was) be executed for each job?

Comment (vyasr) [reply flow/project.py:615]: @klywang A shell command with multiple jobs would be used in the same way that a Python operation with multiple operations would be used. For example, if you want to make a combined plot of all jobs where the statepoint key `a = 1`, the new aggregation feature would let you write a Python operation `plot(jobs)` that takes all the desired jobs and makes a plot. Now if for some reason you need to use a non-Python tool to make a specific kind of plot, you could use a shell command operation that runs `OTHER_TOOL --files job1_fn job2_fn ...`.

@bdice I agree, we should not break existing code. I think your proposal makes sense, we should handle this in as close to the same manner that we handle Python operations as possible. Python operations are called as `op(*jobs)`, which works in both the single-job and many-job contexts. The only difference between what you proposed and an exact adaptation to the Python code is that the format arguments still contains the tuple, not an unpacked set of jobs, but in a format string context that kind of unpacking doesn't make much sense, so passing the tuple in addition to aliasing the single-job case seems right to me.

Comment (bdice) [reply flow/project.py]: Great. I updated this code.

**Review by kidrahahjo — APPROVED** [review]
This looks good to me. Also, It's worth writing tests for the cmd operations when used with aggregation. I will write those tests in the branch that I created for aggregation and push the code.

**Review by tcmoore3 — APPROVED** [review]
LGTM

**Pull request merged**